### DISCUSSION TITLE
Adaptive MPC controller

### DISCUSSION THREAD

**Discussion by icnosvbovwb** [OpenFAST/openfast#2225] [Q&A]
Dear @jjonkman 
![image](https://github.com/OpenFAST/openfast/assets/107483730/8eb20403-e58b-43f3-b63a-c44c4448c35d)
Q1: Is the partial derivative of Tr and Fr derived from a formula? Or is it based on the linearization function? 
Because based on this article "On the Design and Training of Linear Model Predictive Control for Wind Turbines", I am reproducing the MPC controller and currently encountering some code setting issues.

![image](https://github.com/OpenFAST/openfast/assets/107483730/e6329c26-abca-41be-aff4-aaad8c5c85ed)
Q2: How to calculate KK_Tw, KK_Tθ and KK_TV, respectively? I guess it's based on the partial derivative of Tr over W in the picture 2，right?
For example : sensitivity of Torque to the rotor collective blade-pitch angle ∂T/∂w=(-0.5*rho*pi*R^4^v*Cp)/λ^2 at the steady-state operating point of 18m/s wind speed, ∂T/∂v=(rho*pi*R^3^v*Cp)/λ, Right? If this is correct，I don't know how to calculate ∂T/∂θ.
KK_Tw = 10^6*(-10.35) ;
KK_Tθ = 10^6*(-0.89);
KK_TV = 10^6*1.20 ;
Similarly，∂F/∂v=rho*pi*R^2^v*Ct, I don't know how to calculate ∂F/∂w and  ∂F/∂θ.
Partial derivative of Ft at the steady-state operating point of 18m/s wind speed:
KK_Fw = 10^5*(-6.15 );
KK_Fbeta = 10^5*(-0.69);
KK_FV = 10^5*(0.77);

Q3: How to calculate the Damping ratio of tower first fore-aft bending?
ksi_t = 0.062;     

Q4: What are the Ft_op and dt_op, respectively?
V_op = 18 ;  
beta_op = 14.77 ;
lamda_op = 12.1*pi./30*R/18 ;
Ft_op = 10^5*4.347 ;
dt_op = 0.2117;

**Comment by jjonkman** [comment]
Dear @icnosvbovwb,

I'm not familiar with the paper you reference.  Can you share the paper or a link?

Best regards,

**Comment by icnosvbovwb** [comment]
Dear @jjonkman 
[On the design and tuning of linear model predictive control for wind.pdf](https://github.com/OpenFAST/openfast/files/15383344/On.the.design.and.tuning.of.linear.model.predictive.control.for.wind.pdf)

**Reply by jjonkman** [reply]
I have not read this paper myself; perhaps someone more familiar with this paper can respond.

Best regards,
